react-toggle-switch-component/src/toggle-switch.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+export default class ToggleSwitch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(prevState => ({
```

Comment:
while passing the setState function a callback is totally valid. It is typically best to only do so in situations which require it, which are uncommon. For now I would focus on just passing an object in with the property or properties you want to update with their new values. Again, this is not wrong, and you implemented it well! Just passing on some info!